sympy/polys/matrices/domainmatrix.py

```diff
@@ -303,71 +655,114 @@ def pow(A, n):
             return sqrtAn * sqrtAn
 
     def rref(self):
-        """Return reduced row-echelon form of a DomainMatrix and indices of pivot vars.
+        r"""
+        Return reduced-row echelon form and list of pivots for the DomainMatrix
+
+        Returns
+        =======
+
+        (DomainMatrix, list)
+            reduced-row echelon form and list of pivots for the DomainMatrix
 
         Raises
         ======
 
-         ValueError('Not a field')
-            if domain of self is not a field
+        ValueError
+            If the domain of DomainMatrix not a Field
 
         Examples
         ========
 
-        >>> from sympy.polys.matrices import DomainMatrix
         >>> from sympy import QQ
-        >>> A = DomainMatrix([
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> domainmatrix = DomainMatrix([
         ...     [QQ(2), QQ(-1), QQ(0)],
         ...     [QQ(-1), QQ(2), QQ(-1)],
         ...     [QQ(0), QQ(0), QQ(2)]], (3, 3), QQ)
-        >>> rref_matrix, rref_pivots = A.rref()
+
+        >>> rref_matrix, rref_pivots = domainmatrix.rref()
         >>> rref_matrix
         DomainMatrix([[1, 0, 0], [0, 1, 0], [0, 0, 1]], (3, 3), QQ)
         >>> rref_pivots
         (0, 1, 2)
 
+        See Also
+        ========
+
+        :py:meth:`~.lu`
+
         """
         if not self.domain.is_Field:
             raise ValueError('Not a field')
         rref_ddm, pivots = self.rep.rref()
         return self.from_rep(rref_ddm), tuple(pivots)
 
     def nullspace(self):
-        """Returns the Nullspace of a DomainMatrix.
+        r"""
+        Returns the Null Space for the DomainMatrix
+
+        Returns
+        =======
+
+        DomainMatrix
+            Null Space of the DomainMatrix
 
         Examples
         ========
 
+        >>> from sympy import ZZ, QQ
         >>> from sympy.polys.matrices import DomainMatrix
-        >>> from sympy import QQ
-        >>> A = DomainMatrix([[QQ(1), QQ(-1)], [QQ(2), QQ(-2)]], (2, 2), QQ)
-        >>> A.nullspace()
-        DomainMatrix([[1, 1]], (1, 2), QQ)
+        >>> domainmatrix = DomainMatrix([
+        ...    [ZZ(1), ZZ(1)],
+        ...    [ZZ(0), ZZ(1)]], (2, 2), ZZ)
+        >>> dMF = domainmatrix.convert_to(QQ)
+        >>> dMF
+        DomainMatrix([[1, 1], [0, 1]], (2, 2), QQ)
+
+        >>> dMF.nullspace()
+        DomainMatrix([], (0, 2), QQ)
```

Comment:
Let the original example remain as it is. Here also

Reply:
@hyadav2k. The reason I am saying let the examples remain is that you've generally used variable names like dMF, domainmatrix1 etc. and I feel A, B as written by me is a better choice. Obviously, this is a very small issue but I still we should stick to A & B